scripts/in_container/prod/entrypoint_prod.sh

```diff
@@ -83,24 +84,12 @@ function wait_for_connection {
     # It tries `CONNECTION_CHECK_MAX_COUNT` times and sleeps `CONNECTION_CHECK_SLEEP_TIME` between checks
     local connection_url
     connection_url="${1}"
-    local detected_backend=""
-    local detected_host=""
-    local detected_port=""
-
-    # Auto-detect DB parameters
-    # Examples:
-    #  postgres://YourUserName:password@YourHostname:5432/YourDatabaseName
-    #  postgres://YourUserName:password@YourHostname:5432/YourDatabaseName
-    #  postgres://YourUserName:@YourHostname:/YourDatabaseName
-    #  postgres://YourUserName@YourHostname/YourDatabaseName
-    [[ ${connection_url} =~ ([^:]*)://([^:@]*):?([^@]*)@?([^/:]*):?([0-9]*)/([^\?]*)\??(.*) ]] && \
-        detected_backend=${BASH_REMATCH[1]} &&
-        # Not used USER match
-        # Not used PASSWORD match
-        detected_host=${BASH_REMATCH[4]} &&
-        detected_port=${BASH_REMATCH[5]} &&
-        # Not used SCHEMA match
-        # Not used PARAMS match
+    local detected_backend
+    detected_backend=$(python -c "from urllib.parse import urlsplit; print(urlsplit('${connection_url}').scheme)")
```

Comment:
```suggestion
    detected_backend=$(python -c "from urllib.parse import urlsplit; import sys; print(urlsplit(sys.argv[1]).scheme)" "${connection_url}")
```
I am concerned that some inputs may lead to Command Injection and therefore incorrect script behavior.

Reply:
For readability, we can extraactt it to a separate function.
```bash
function get_url_part() {
    url_part="$1";
    connection_url="$2";
    python -c "from urllib.parse import urlsplit; import sys; print(getattr(urlsplit(sys.argv[2]), sys.argv[1]))" "${url_part}" "${connection_url}"
}
get_url_part scheme redis://
```

Reply:
I prefer to leave it explicit. I think it is far less readable with url_part parameter to be honest.

Reply:
adding separate function is what I tried initially, and I REALLY found it unnecessary

Reply:
> I am concerned that some inputs may lead to Command Injection and therefore incorrect script behavior.

Yep. Agree on that one.